### PR TITLE
Updated rooster command bar based on new design

### PR DESCRIPTION
**Pull request by adriantran**
Updated rooster command bar based on new design:
- image insert
- emjoi
- font color
- highlight
- link

Also added Callout hooks for use with FocusOutShell, various small fixes improvements, and changed LeanRooster base to React.Component to avoid warnings
(color icons to come later)

### CODE REVIEW

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    
+    .rooster-command-bar-color-cell {
+        width: 20px;
+        height: 20px;
+        margin: 4px;
+        border-style: solid;
+        border-width: 2px;
+        border-color: transparent;
+
+        @media screen and (-ms-high-contrast: active) {
```
Comment (pranavgk) [inline packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.scss]: This works with other browsers as well right?

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx [hunk 2]
```diff
@@ -1,38 +1,30 @@
 import * as React from 'react';
 import { Async, css } from 'office-ui-fabric-react/lib/Utilities';
-// TODO-1 import { CommandBar, ICommandBarItemProps } from '@uifabric/experiments/lib/CommandBar';
```
Comment (pranavgk) [inline packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx]: Not planning to use the experimental command bar anymore?

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx [hunk 3]
```diff
@@ -43,24 +35,25 @@ export default class RoosterCommandBar extends React.Component<RoosterCommandBar
...
-                    // TODO-1 update experimental OfficeFabric CommandBar
-                    /* overflowItemProps={{ menuClassName: calloutClassName, onAfterMenuDismiss: calloutOnDismiss, } as IOverflowItemProps} */
-                    className={css("command-bar")}
+                    className={"command-bar"}
                     items={this._buttonKeys
-                        .map(key => this._getMenuItem(OOB_COMMAND_BAR_ITEM_MAP[key]))
+                        .map(key => this._getMenuItem(OutOfBoxCommandBarItemMap[key]))
                         .filter(menuItem => !!menuItem)}
                 />
+                <input type="file" ref={this._fileInputOnRef} accept="image/*" style={DisplayNoneStyle} onChange={this._fileInputOnChange} />
```
Comment (pranavgk) [inline packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx]: nit: inline?

Comment (pranavgk) [reply packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx]: I mean inline ref btw, dunno why GitHub doesnt highlight the selection :(

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.scss (hunk 1 above)
Comment (adriantran) [reply packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.scss]: This is specifically for Edge and IE. It's the same pattern as Office Fabric and VSSUI.

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx (hunk 2 above)
Comment (adriantran) [reply packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx]: nope, even the owner doesn't know when it'll ship

File: packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx (hunk 3 above)
Comment (adriantran) [reply packages/roosterjs-react-command-bar/lib/components/RoosterCommandBar.tsx]: it's less performant if you inline, the delegate is created each time render is called

File: packages/roosterjs-react-common/lib/utils/ImageManager.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        }
+
+        // note: add ID for debug/identification (possibly handle undo/redo scenarios later)
+        let preserveId = true;
+        if (!placeholder.id) {
+            preserveId = false;
+            placeholder.id = `paste-image-placeholder-${ImageManager.Id++}`;
+        }
+
+        let reader = new FileReader();
```
Comment (pranavgk) [inline packages/roosterjs-react-common/lib/utils/ImageManager.ts]: Looks like in VSTS we use a different rest endpoint (attachments) for uploading images using the toolbar, and for pasted images we use uploadAttachmentBinary (MVC) endpoint.
The attachments rest endpoint actually just needs the file object and not the dataURL:
image

I can write a different implementation of ImageManager, but in that case we would be nice to expose the spinner SVG.

**Pull request merged**